connector/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/consumer/KafkaDataConsumer.scala

```diff
@@ -548,7 +572,10 @@ private[kafka010] class KafkaDataConsumer(
       fetchedData: FetchedData,
       offset: Long,
       pollTimeoutMs: Long): Unit = {
-    val (records, offsetAfterPoll, range) = consumer.fetch(offset, pollTimeoutMs)
+    val (records, offsetAfterPoll, range) = timeNanos {
+      consumer.fetch(offset, pollTimeoutMs)
+    }
+    numPolls += 1
```

Comment:
I'd say we need to count the number of records here because we are focusing on the interaction with Kafka, which effectiveness of the cache on fetched data should not contribute.

Reply:
I was thinking about it too. However, `records` is a List and retrieving number of records here may not be very cheap.

Reply:
Oh wait, maybe it's not. Let me try.

Reply:
There is no guarantee that the implementation of List will provide size as O(1), but most implementations are happy to spare slight overhead cost to track the total number of elements, since size has been one of most calling methods in List. (Implementations in Java standard library are mostly following)

Please update this and let me know, I guess this is the last one remaining.